File: PycharmProjects/Store_Herencia/py/SystemStore.py



class SystemStore:
    secure_password = "thisisnotasecurepassword"

    def __init__(self):
        self.store = None

    def run_program(self):
        valid_password = False

        while not valid_password:
            print("\n*** WELCOME ***")
            password = input("Enter the password please: ")

            if password == self.secure_password:
                valid_password = True
                self.run_system_menu()
            else:
                print("\nIncorrect password, try again")

    def run_system_menu(self):
        opcion = 0
        while opcion != 11:
            print("\n** WELCOME TO THE SYSTEM **")
            print("1. Register Product")
            print("2. Register Client")
            print("3. Show Products")
            print("4. Show Clients")
            print("5. Delete Product")
            print("6. Delete Client")
            print("7. Add to Stock")
            print("8. Remove from Stock")
            print("9. Make Purchase")
            print("10. View Purchases")
            print("11. Exit")
            opcion = int(input("Select your option: "))

            if opcion == 1:
                print("\nYou selected register product")
                print("Enter the type of product to register")
                print("1. Cleaning")
                print("2. Food")
                print("3. Householde Appliance")
                print("4. Makeup")
                opcion_producto = int(input("Select your option: "))

                if opcion_producto == 1:
                    self.store.register_cleaning_product()
                elif opcion_producto == 2:
                    self.store.register_food_product()
                elif opcion_producto == 3:
                    self.store.register_household_appliance_product()
                elif opcion_producto == 4:
                    self.store.register_beauty_product()

            elif opcion == 2:
                self.store.register_client()

            elif opcion == 3:
                self.store.show_all_products()

            elif opcion == 4:
                self.store.show_all_clients()

            elif opcion == 5:
                serial_number = int(input("Enter the serial number: "))
                self.store.remove_product_by_serial_number(serial_number)

            elif opcion == 6:
                client_id = input("Enter the ID of the client to remove: ")
                self.store.remove_client_by_id(client_id)

            elif opcion == 7:
                serial_number = int(input("Enter the serial number of the product: "))
                quantity_to_add = int(input("Enter the quantity to add to the stock: "))
                self.store.add_to_stock_by_serial_number(serial_number, quantity_to_add)

            elif opcion == 8:
                serial_number = int(input("Enter the serial number of the product: "))
                quantity_to_remove = int(input("Enter the quantity to remove from the stock: "))
                self.store.remove_from_stock_by_serial_number(serial_number, quantity_to_remove)

            elif opcion == 9:
                client_id = input("Enter your client ID: ")
                client = self.store.find_client_by_id(client_id)
                if client is None:
                    print("Client with the given ID not found.")
                else:
                    self.store.purchase_product(client)

            elif opcion == 10:
                self.store.view_purchase_history()

        print("Leaving the store, good bye!")

File: OneDrive/Documentos/Store_Herencia/Store_Herencia/src/store/herencia/SystemStore.java
package store.herencia;

import java.util.Scanner;

public class SystemStore {
    private final String secure_password = "thisisnotasecurepassword";
    private Scanner sc = new Scanner(System.in);
    private Store store = new Store();

    public void runProgram() {
        boolean validPassword = false;

        do {
            System.out.println("\n*** WELCOME ***");
            System.out.println("Enter the password please: ");
            String password = sc.nextLine();

            if (password.equals(secure_password)) {
                validPassword = true;
                runSystemMenu();
            } else {
                System.out.println("\nIncorrect password, try again");
            }
        } while (!validPassword);
    }

    private void runSystemMenu() {
        int opcion = 0;
        do {
            System.out.println("\n** WELCOME TO THE SYSTEM **");
            System.out.println("1. Register Product");
            System.out.println("2. Register Client");
            System.out.println("3. Show Products");
            System.out.println("4. Show Clients");
            System.out.println("5. Delete Product");
            System.out.println("6. Delete Client");
            System.out.println("7. Add to Stock");
            System.out.println("8. Remove from Stock");
            System.out.println("9. Make Purchase");
            System.out.println("10. View Purchases");
            System.out.println("11. Exit");
            System.out.println("Select your option: ");
            opcion = sc.nextInt();
            switch (opcion) {
                case 1:
                    System.out.println("\nYou selected register product");
                    System.out.println("Enter the type of product to register");
                    System.out.println("1. Cleaning");
                    System.out.println("2. Food");
                    System.out.println("3. Householde Appliance");
                    System.out.println("4. Makeup");
                    System.out.println("Select your option: ");
                    int opcionProcucto = sc.nextInt();
                    
                    switch (opcionProcucto) {
                        case 1:
                        store.registerCleaningProduct();
                            break;
                        case 2:
                        store.registerFoodProduct();
                            break;
                        case 3:
                        store.registerHouseholdApplianceProduct();
                            break;
                        case 4:
                        store.registerBeautyProduct();
                    }
                    break;
                case 2:
                    store.registerClient();
                    break;
                case 3: 
                    store.showAllProducts();
                    break;
                case 4:
                    store.showAllClients();
                    break;
                case 5:
                    System.out.println("Enter the serial number: ");
                    int serialNumber1 = sc.nextInt();
                    store.removeProductBySerialNumber(serialNumber1);
                    break;
                case 6:
                   System.out.println("Enter the ID of the client to remove:");
                   String clientId = sc.nextLine();
                   store.removeClientById(clientId);
                break;
                case 7:
                System.out.println("Enter the serial number of the product:");
                int serialNumber2 = sc.nextInt();
                sc.nextLine(); 
               
                System.out.println("Enter the quantity to add to the stock:");
                int quantityToAdd = sc.nextInt();
                sc.nextLine(); 
                
                store.addToStockBySerialNumber(serialNumber2, quantityToAdd);
                break;
                case 8: 
                System.out.println("Enter the serial number of the product:");
                int serialNumber3 = sc.nextInt();
                sc.nextLine(); 
               
                System.out.println("Enter the quantity to add to the stock:");
                int quantityToRemove = sc.nextInt();
                sc.nextLine(); 
                store.removeFromStockBySerialNumber(serialNumber3, quantityToRemove);
                break;
                case 9:
                System.out.println("Enter your client ID:");
                String clientId2 = sc.nextLine();

                Client client = store.findClientByID(clientId2);
                if (client == null) {
                    System.out.println("Client with the given ID not found.");
                    break;
                }

                store.purchaseProduct(client);
                break;
                case 10:
                store.viewPurchaseHistory();
                break;
            }
        } while(opcion != 11);
        System.out.println("Leaving the store, good bye!");
    }
}
